app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -17,24 +17,41 @@ import kotlin.math.roundToInt
 import org.opencv.core.Size as CvSize
 
 class DroidCvPattern(
-    private var mat: Mat? = Mat(),
+    private var mat: Mat = Mat(),
+    private var alpha: Mat? = null,
     private val ownsMat: Boolean = true
 ) : IPattern {
+    private data class MatWithAlpha(val mat: Mat, val alpha: Mat?)
+
     private companion object {
-        fun makeMat(stream: InputStream): Mat {
+        fun makeMat(stream: InputStream): MatWithAlpha {
             val byteArray = stream.readBytes()
+            MatOfByte(*byteArray).use {
+                val grayScale = Imgcodecs.imdecode(it, Imgcodecs.IMREAD_GRAYSCALE)
```

Comment:
Add a comment here that imread grayscale and converting to grayscale from imread unchanged is different, or someone might do that again.

Other than that, tested that works on both NA and JP on wide screen.

Reply:
Also, make sure to mention alpha matching in the commit message.